src/routes/login_with_microsoft/microsoftAuth.ts

```diff
@@ -0,0 +1,44 @@
+import Router from "@koa/router";
+import base64url from "base64-url";
+import config from "config";
+import passport from "koa-passport";
+import { Strategy, StrategyOptions, VerifyCallback } from "passport-oauth2";
+import { findOrCreateUser } from "../../data/users";
+
+export const baseURL = process.env.NODE_ENV === "production" ? "https://goty.app" : "http://localhost:8080";

```

Comment:
notice localhost:8080 here instead of localhost.shea.at like in google auth. this is because, while google disallows localhost but allows http, microsoft allows localhost and disallows http (except on localhost). this is annoying.

it's mostly harmless though; if you try to login from localhost.shea.at, you'll be transparently redirected to localhost.